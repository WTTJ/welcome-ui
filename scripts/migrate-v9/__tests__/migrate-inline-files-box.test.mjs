import { readFileSync } from 'fs'
import { dirname, resolve } from 'path'
import { fileURLToPath } from 'url'

import { describe, expect, it } from 'vitest'

import { copyDirSync, deleteDirRecursive } from '../helpers/file-utils.mjs'
import { migrateAll } from '../index.mjs'

const __dirname = dirname(fileURLToPath(import.meta.url))

describe('Inline Migration - Box Components', () => {
  it('migrates Box components with various props to div with classNames', async () => {
    const fixturesDir = resolve(__dirname, './__fixtures__')
    const tempDir = resolve(__dirname, '../temp/inline-box-test')

    // Clean up and copy fixtures
    deleteDirRecursive(tempDir)
    copyDirSync(fixturesDir, tempDir)

    // Run migration
    await migrateAll(tempDir, { copyDir: false, interactive: false })

    // Read results
    const migratedComponent = readFileSync(resolve(tempDir, 'BoxComponent.tsx'), 'utf8')

    // Verify with snapshots
    expect(migratedComponent).toMatchSnapshot('box-component.tsx')

    // Cleanup
    deleteDirRecursive(tempDir)
  })
})
