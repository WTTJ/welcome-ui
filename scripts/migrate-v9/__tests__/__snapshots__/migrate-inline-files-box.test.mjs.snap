// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Inline Migration - Box Components > migrates Box components with various props to div with classNames > box-component.tsx 1`] = `
"import { Box } from 'welcome-ui/Box'

export const BoxComponent = () => {
  return (
    <div>
      <div className="bg-primary-500 flex mt-lg">Main content</div>
      <div className="border border-neutral-200 p-md rounded-sm">Card content</div>
      <section className="max-w-container" mx="auto">
        Container content
      </section>
    </div>
  )
}
"
`;
